### DISCUSSION TITLE
A necessary tweak for the NPC Aggression Timer plugin

### DISCUSSION THREAD

**Discussion by Elite-Mercenary-Raven** [runelite/runelite#16411] [Ideas]
(This post aims to continue the discussion in #15078, offering a streamlined solution.)

The problem:

Currently, if I have the NPC Aggression Timer plugin activated, every time I log in there's an annoying "help" message that appears, covering a large portion of the screen. It gets in the way and creates unnecessary clutter. Like most people I'm sure, I don't want to go out of my way to set up the plugin every time I play (immediately teleporting and running 10 tiles, right after logging in), but I also don't want something unnecessary taking up space and blocking my vision. It's also unlikely that what I'm doing will allow the plugin to passively set itself up at the very beginning of my session, clearing the big block of annoying "help" text. When I log in I often want to continue the grind I was doing when I last logged off. I'm sure that's a common desire for most people, and it applies to a lot of things you would do in the game: bank standing, fishing, WCing, mining, smithing, etc. These all easily lend themselves to this desire; so typically it's best to keep the plugin off, even if you do like to use it. I'm sure a lot of other people keep it off now too.

There are times when the plugin is useful, but being forced to toggle it on only when I need it creates another problem: now every time I want to use it, not only do I have to go out of my way to open my configuration menu, find the plugin, and then toggle it on, I also have to go out of my way to set up the plugin manually in-game, something that in many cases I already would have done earlier without even thinking about it, if the plugin had already been on. Setting it up like this over and over is a measurable waste of time that I'm sure others like me have gone through repeatedly, and I know this process could be avoided if the plugin functioned a little differently.

Unfortunately--assuming you don't want an unnecessary block of text on your screen every time you log in--the best way to use the NPC Aggression Timer plugin right now is to keep it off 99% of the time. It doesn't matter if it's able to work automatically, because more often than not it’s going to be more annoying than it is helpful.

---

This is the most user-friendly way I can think of to fix this issue, so using the plugin can be enjoyed in a way that is as effortless and clutter-free as possible. (I'm not a programmer familiar with RuneLite, so I don't know if any of these points have to be adapted a bit to work; but from my experience, they seem doable.) 

1) The timer infobox needs a second, visually distinct mode for when the plugin is not yet calibrated, to be used instead of the large block of help text. When mousing over the infobox in this mode, only then should it tell you how to calibrate the plugin. This would reduce visual clutter and be a more streamlined design. For clear visual communication, the normal icon should be edited here, appearing in black and white. It may also be worth superimposing either an "!", "?", or some kind of "RuneScapey" prompting icon, in a way that doesn't feel nagging or too attention-drawing (no bright colors, no flashing, glowing, or moving text, etc). It should be up to the player to choose to go out of their way to calibrate it.

2) For people who like to use the NPC Aggression Timer plugin without the "Always active" option checked, there should be an option to hide this "not yet calibrated" infobox while relevant NPCs aren't around (so it only appears when the plugin thinks you're trying to use it). This would help reduce visual clutter.

3) There could also be a universal toggle to never show the calibration infobox, that works regardless of if "Always active" is checked. This would be disabled by default. 

That should solve every problem the plugin has right now: removing the need for a big text box, maintaining an indicator for whether or not the plugin is able to function properly (using a simple, intuitive graphic to communicate rather than a clunky block of text), and unintrusively informing players of how to set it up.

**Comment by Nightfirecat** [comment]
The plugin saves your last location when logging out/closing the client, so this help message should not be re-appearing constantly for you. Are you switching between multiple clients frequently?

**Reply by Elite-Mercenary-Raven** [reply]
That's a good point. It's possible that it's happening because I run a few alt accounts. I do only use the Jagex launcher. It might even be more likely that the issue is because I play on 2 different computers, one Mac, and one PC, so there might be some de-sync there. I'm not sure when the RL client sends everything to the cloud to back up, but I don't always close my client between sessions. (Edit: Ignore everything I said about using 2 computers. I just checked and didn't realize how long it's been since I played on Mac).

**Reply by Elite-Mercenary-Raven** [reply]
Even if it's not intended that that block of help text appears every session, I do still think it takes way too much screen real estate, considering that text is only helpful on the specific condition that I'm trying to use the plugin immediately when I log in. I love the plugin when I need it, but in other scenarios, I would love to have the freedom to ignore it. :)

**Reply by Elite-Mercenary-Raven** [reply]
Update:

I've been playing a lot more, recently, and only on one account on one computer. Every single time I log in I get that help text. Do you know a way to fix this issue? I need it to work in a "set it and forget it" way. I don't want to have to micromanage the plugin, toggling it on and off, and configuring it all the time. Thank you!

**Reply by Im-Zoo** [reply]
I agree the help message takes up too much of the screen. @Elite-Mercenary-Raven do you happen to log onto mobile between sessions on your computer? Would that be what causes the help message to appear? 

@Nightfirecat, could you make a toggle for the help message to be the aggression icon with an exclamation point rather than the countdown time to indicate that you have to teleport away to make it work again?
